### ISSUE TITLE
[Bug]: Upon successful authentication, Shopify-cli opens a page with the URL http://127.0.0.1:9292/password?_fd=0 and doesn't render main content

### ISSUE THREAD

**Issue by merziy** [Shopify/cli#2186]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme, Other

### Expected behavior

After successfully authenticating to a store with `shopify theme dev --store https://<STORE NAME>/`, it opens up a new page with the URL http://127.0.0.1:9292/ and renders the code.

### Actual behavior

After successfully authenticating to a store with `shopify theme dev --store https://<STORE NAME>/`, it opens up a new page with the URL http://127.0.0.1:9292/password?_fd=0 and an empty `div.shopify-section`. It does render other sections, like the nav, header, and footer, but never the main content/Shopify sections on pages.

This does not happen in all of my projects, which all have passwords, but it does on this one, which I also have open a window with my account logged in as a collaborator. I developed the store under a staff account until I transferred ownership to the client and they removed me as a staff and added me as a collaborator under the same email address. 

### Verbose output

```shell
@mac ➜ karee-hays-esthetics-final rvm:(ruby-3.2.2) git:(main) shopify theme dev --store https://kareehays.myshopify.com/ --verbose
2023-06-15T13:25:38.954Z: Running command theme dev
  shopify:theme:dev init version: @oclif/core@2.1.4 argv: [ '--store', 'https://kareehays.myshopify.com/', '--verbose' ] +0ms
2023-06-15T13:25:38.969Z: Ensuring that the user is authenticated with the Theme API with the following scopes:
[]

2023-06-15T13:25:38.969Z: Ensuring that the user is authenticated with the Admin API with the following scopes for the store kareehays.myshopify.com:
[]

2023-06-15T13:25:38.969Z: Getting session store...
2023-06-15T13:25:38.970Z: Validating existing session against the scopes:
[
  "openid",
  "https://api.shopify.com/auth/shop.admin.graphql",
  "https://api.shopify.com/auth/shop.admin.themes",
  "https://api.shopify.com/auth/partners.collaborator-relationships.readonly",
  "https://api.shopify.com/auth/shop.storefront-renderer.devtools",
  "https://api.shopify.com/auth/partners.app.cli.access"
]
For applications:
{
  "adminApi": {
    "scopes": [],
    "storeFqdn": "kareehays.myshopify.com"
  }
}

2023-06-15T13:25:38.970Z: Sending Identity Introspection request to URL: https://accounts.shopify.com/oauth/introspection
2023-06-15T13:25:38.971Z: Sending POST request to URL https://accounts.shopify.com/oauth/introspection
With request headers:
 - User-Agent: Shopify CLI; v=3.46.5
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-15T13:25:39.103Z: Request to https://accounts.shopify.com/oauth/introspection completed in 128 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"1fae84ddf0a70ac4d0541ce0f0c3aaa9"
 - x-request-id: f2383b7a-5e20-45ef-a1cb-9d468794126e

2023-06-15T13:25:39.105Z: The identity token is valid: true
2023-06-15T13:25:39.105Z:
The validation of the token for application/identity completed with the following results:
- It's expired: false
- It's invalid in identity: false

2023-06-15T13:25:39.105Z: The current session is valid but needs refresh. Refreshing...
2023-06-15T13:25:39.106Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=refresh_token&access_token=atkn_2ecdaa7a58ec7d68bdcc54947c95e728588b3c707aec771769f2f8d2452fae8a&refresh_token=atkn_2e243dbf6bc931fe64d88828b37606da07e2c82ea2a3a72486813a56801525b5&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3
With request headers:
 - User-Agent: Shopify CLI; v=3.46.5
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-15T13:25:39.321Z: Request to https://accounts.shopify.com/oauth/token?grant_type=refresh_token&access_token=atkn_2ecdaa7a58ec7d68bdcc54947c95e728588b3c707aec771769f2f8d2452fae8a&refresh_token=atkn_2e243dbf6bc931fe64d88828b37606da07e2c82ea2a3a72486813a56801525b5&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3 completed in 214 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"7820c0b7a593bfa4e992087916dd3a7e"
 - x-request-id: 7d181d6b-4a37-4847-ab79-250b3e4307e0

2023-06-15T13:25:39.324Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=271e16d403dfa18082ffb3d197bd2b5f4479c3fc32736d69296829cbb28d41a6&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.app.cli.access&subject_token=atkn_195c66a0a51050409b36d00b8a7f773ba689566610580da2b0b65829f3bd0ef2
With request headers:
 - User-Agent: Shopify CLI; v=3.46.5
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-15T13:25:39.324Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=ee139b3d-5861-4d45-b387-1bc3ada7811c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.storefront-renderer.devtools&subject_token=atkn_195c66a0a51050409b36d00b8a7f773ba689566610580da2b0b65829f3bd0ef2
With request headers:
 - User-Agent: Shopify CLI; v=3.46.5
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-15T13:25:39.461Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=ee139b3d-5861-4d45-b387-1bc3ada7811c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.storefront-renderer.devtools&subject_token=atkn_195c66a0a51050409b36d00b8a7f773ba689566610580da2b0b65829f3bd0ef2 completed in 135 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"ebe97e4ab21fb8432780f7a0da3291e6"
 - x-request-id: abfd293e-67d6-42ed-a931-34a272f1671d

2023-06-15T13:25:39.716Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=271e16d403dfa18082ffb3d197bd2b5f4479c3fc32736d69296829cbb28d41a6&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.app.cli.access&subject_token=atkn_195c66a0a51050409b36d00b8a7f773ba689566610580da2b0b65829f3bd0ef2 completed in 390 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"3642ff7412c8d5b5ea2ef4535dcd87c9"
 - x-request-id: d4532fb6-07c1-4b15-9f70-116f86d1c102

2023-06-15T13:25:39.718Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=7ee65a63608843c577db8b23c4d7316ea0a01bd2f7594f8a9c06ea668c1b775c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.graphql+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.themes+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.collaborator-relationships.readonly&subject_token=atkn_195c66a0a51050409b36d00b8a7f773ba689566610580da2b0b65829f3bd0ef2&destination=https%3A%2F%2Fkareehays.myshopify.com%2Fadmin
With request headers:
 - User-Agent: Shopify CLI; v=3.46.5
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-15T13:25:39.853Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=7ee65a63608843c577db8b23c4d7316ea0a01bd2f7594f8a9c06ea668c1b775c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.graphql+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.themes+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.collaborator-relationships.readonly&subject_token=atkn_195c66a0a51050409b36d00b8a7f773ba689566610580da2b0b65829f3bd0ef2&destination=https%3A%2F%2Fkareehays.myshopify.com%2Fadmin completed in 134 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"aaffb41f33709f921b77f9e08773bd8c"
 - x-request-id: 191349bc-72a0-4972-8c6f-48dcb7b64cb9

2023-06-15T13:25:39.855Z: Setting session store...
2023-06-15T13:25:39.865Z: Ensuring that the user is authenticated with the Storefront API with the following scopes:
[]

2023-06-15T13:25:39.865Z: Getting session store...
2023-06-15T13:25:39.866Z: Validating existing session against the scopes:
[
  "openid",
  "https://api.shopify.com/auth/shop.admin.graphql",
  "https://api.shopify.com/auth/shop.admin.themes",
  "https://api.shopify.com/auth/partners.collaborator-relationships.readonly",
  "https://api.shopify.com/auth/shop.storefront-renderer.devtools",
  "https://api.shopify.com/auth/partners.app.cli.access"
]
For applications:
{
  "storefrontRendererApi": {
    "scopes": []
  }
}

2023-06-15T13:25:39.866Z: Sending Identity Introspection request to URL: https://accounts.shopify.com/oauth/introspection
2023-06-15T13:25:39.866Z: Sending POST request to URL https://accounts.shopify.com/oauth/introspection
With request headers:
 - User-Agent: Shopify CLI; v=3.46.5
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-15T13:25:39.970Z: Request to https://accounts.shopify.com/oauth/introspection completed in 102 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"e491dbfd81f8e27981ed2279f8a11ae7"
 - x-request-id: d448e574-da2d-4d30-87fd-2f1e09c9eb0b

2023-06-15T13:25:39.972Z: The identity token is valid: true
2023-06-15T13:25:39.973Z:
The validation of the token for application/identity completed with the following results:
- It's expired: false
- It's invalid in identity: false

2023-06-15T13:25:39.981Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/ruby -v
  · Working directory: /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final

2023-06-15T13:25:40.002Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle -v
  · Working directory: /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final

2023-06-15T13:25:40.088Z: Reading the content of file at /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile...
2023-06-15T13:25:40.089Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle config set --local path /Users/alexandermathis/Library/Caches/shopify-gems-nodejs
  · Working directory: /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-15T13:25:40.167Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle config set --local without development:test
  · Working directory: /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-15T13:25:40.246Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle install
  · Working directory: /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-15T13:25:40.394Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle exec /opt/homebrew/opt/ruby/bin/ruby /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify theme token --admin atkn_eyJhbGciOiJSUzI1NiIsImtpZCI6Ik4teW1MUDR5TWt2MlR2X3hjWkNNSlltZ2JDbjhQbm5WLWRBbUJHbTlHb3cifQ.eyJ0IjoiRVR5TnFYS2FDL1FJNzdWelg5SDc2N3hUVlkramkxdHBlazBiR3J1NFRKbz0ifQ.Y33owdE8jfrEfXTub6UK6nwYt55l6jC7LXNAO5BgP8qMIQfNeKGfql90oXli_8O-5IvFAra7R5RoNprzKKh5lxl9QRzHyzlo778tRBMAHkVG0_dHduwqLtPuVYIg613abJNAyVcbibLytWdUHLxY-xwatoNZBR4ON7zJOf7LTgSe7PNRFwLw6lzco_uPYcxzNDSrdNI1E6hFl6Q4Uwe210G70D4rphAqhcxmYemmICGdwJhf9zNy8anjTm3WRlp-qVX9pCW5TT0QP_aOOYHeFdiG6CM7QDgbiLNfI1V3DDrGm8qRUPZwHj0A5OX-LGkEvEX945-L-mNO088y3savPg --sfr atkn_eyJhbGciOiJSUzI1NiIsImtpZCI6Ik4teW1MUDR5TWt2MlR2X3hjWkNNSlltZ2JDbjhQbm5WLWRBbUJHbTlHb3cifQ.eyJ0IjoibnRSbnNBeGExc080ellNL3hlTEk4YTBBajJCVGhvUGlKVThwTzJPWkhJOD0ifQ.M9kdXUN0-IeRuSFIOnWLF74DDqjsfsbiUTEA9tHS4qfgrOG4qx9n71h9WhSR6slCn_2A7b7sY7N1-FoCxMMtD_B9PSYYoWmLMa5waBZr4c459xqA85Ky4wpvWJbu8UlbN9jc_s70cPKQNu3HiB3924MtxGbXBLRt9zHG-JpUPM8vpIkCS5NmtMu0K4r9-T3XrLUCw5Ml-VoXo1c7InnJTp_fsfMC0ciewYcAhEG2uCjiHV7fMlry9m_dDcfwmKtI1AZXHnonPSqGjhQH_7umEVG-ghcOwIsDvEU2j4HGm1kFtXF76vnJx1-tiOcgYUf2Cb7vzwN-wIFVCuX9AiFCKA
  · Working directory: /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final

2023-06-15T13:25:40.631Z: Getting development theme...
2023-06-15T13:25:40.869Z: Request to https://kareehays.myshopify.com/admin/api/unstable/themes/151902224675.json?fields=id completed in 236 ms
With response headers:
 - content-type: application/json; charset=utf-8
 - x-request-id: 354f44a9-86f5-42c7-b0e9-dff60cacde00

╭─ success ────────────────────────────────────────────────────────────────────╮
│                                                                              │
│  Preview your theme                                                          │
│    • http://127.0.0.1:9292                                                   │
│                                                                              │
│  Next steps                                                                  │
│    • Customize your theme at the theme editor                                │
│    • Share your theme preview                                                │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯

2023-06-15T13:25:40.903Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/ruby -v
  · Working directory: /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final

2023-06-15T13:25:40.912Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle -v
  · Working directory: /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final

2023-06-15T13:25:40.991Z: Reading the content of file at /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile...
2023-06-15T13:25:40.992Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle config set --local path /Users/alexandermathis/Library/Caches/shopify-gems-nodejs
  · Working directory: /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-15T13:25:41.071Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle config set --local without development:test
  · Working directory: /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-15T13:25:41.149Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle install
  · Working directory: /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-15T13:25:41.297Z:
Running system process:
  · Command: /opt/homebrew/opt/ruby/bin/bundle exec /opt/homebrew/opt/ruby/bin/ruby /opt/homebrew/Cellar/shopify-cli/3.46.5/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify theme serve /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final --live-reload hot-reload --theme 151902224675
  · Working directory: /Users/alexandermathis/dev/juicyorange/karee-hays-esthetics-final

DEBUG GET https://kareehays.myshopify.com/admin/api/unstable/themes.json with request_id: ae2ca975-21aa-4ca0-ad29-bd46978385e5
* Syncing theme #151902224675 on kareehays.myshopify.com
DEBUG GET https://kareehays.myshopify.com/admin/api/unstable/themes/151902224675/assets.json with request_id: 0c2d545d-4fa6-4142-b888-652b607e126c
                                                                            100%
[2023-06-15 09:25:42] INFO  WEBrick 1.7.0
[2023-06-15 09:25:42] INFO  ruby 3.2.2 (2023-03-30) [arm64-darwin22]
[2023-06-15 09:25:42] INFO  WEBrick::HTTPServer#start: pid=17414 port=9292
DEBUG Refreshing preview _secure_session_id cookie
DEBUG Proxying HEAD https://kareehays.myshopify.com/?preview_theme_id=151902224675&_fd=0&pb=0
DEBUG `-> 302 request_id: 137b4e82-9c4f-4962-82c0-f00ed65157a1
DEBUG Proxying GET https://kareehays.myshopify.com/?_fd=0&pb=0
DEBUG `-> 302 request_id: 7fa6f4a7-017f-415c-a1ba-c2fa2cb7aaa4
DEBUG Proxying GET https://kareehays.myshopify.com/password?_fd=0&_fd=0&pb=0
DEBUG `-> 200 request_id: f442649d-b0e4-4446-aebe-9c5575500be2
DEBUG [HotReload] Connected to SSE stream
```

### Reproduction steps

1. Successfully connect to a store with `shopify theme dev --store https://<STORE NAME>/` while being logged in as a collaborator to the same store
2. Notices when you make changes in the code, the files sync
3. Notice parts of every page load, such as the header, nav, and footer, but never the main content

### Operating System

Ventura 13.4 (22F66)

### Shopify CLI version (check your project's `package.json` if you're not sure)

Current Shopify CLI version: 3.46.5

### Shell

zsh

### Node version (run `node -v` if you're not sure)

v20.3.0

**Comment by merziy** [comment]
Running into the same issue with the same project on the same OS with another computer, and on a Windows 11 + WSL (Ubuntu) setup.

**Comment by Mizou04** [comment]
from the docs : "Collaborator accounts give you access to only the sections of a store that a merchant wants you to access, and don't count toward a store's staff limit...".
are you sure the merchant grants you the necessary permissions?

**Comment by kitkatpapi** [comment]
I have done the update and im still having this same issue. any solutions?

**Comment by Khreno** [comment]
Same issue here. I was using Shopify CLI 3.48.0 and upgraded to 3.48.1
I have also cleared npm cache - Reinstalled CLI .... Upgrading my OS to Ventura 13.5 and nothing.


Doing further investion I found that it is a 302 on the network tab of the inspector, and redirects to 127.0.0.1:9292/password

It seems to me that there is an authentication issue to be fixed on Shopify side. 

The only way to bypass this problem for us is to untick the "Restrict store access" by password. However this site is on development and it is not a good fix really.
